huaweicloud/services/aom/resource_huaweicloud_aom_multi_account_aggregation_rule.go

```diff
@@ -0,0 +1,337 @@
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"instance_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"accounts": {
+				Type:     schema.TypeSet,
+				Required: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"name": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"urn": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"join_method": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"joined_at": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+					},
+				},
+			},
+			"services": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"service": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"metrics": {
+							Type:     schema.TypeSet,
+							Required: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+						},
+					},
+				},
+			},
+			"send_to_source_account": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Computed: true,
+			},
+			"enterprise_project_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func buildHeaders(d *schema.ResourceData) map[string]string {
+	moreHeaders := map[string]string{
+		"Content-Type": "application/json",
+	}
+	if v, ok := d.GetOk("enterprise_project_id"); ok {
+		moreHeaders["Enterprise-Project-Id"] = v.(string)
+	}
+	return moreHeaders
+}
+
+func resourceMultiAccountAggregationRuleCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+	client, err := cfg.NewServiceClient("aom", region)
+	if err != nil {
+		return diag.Errorf("error creating AOM client: %s", err)
+	}
+
+	createHttpUrl := "v1/{project_id}/aom/aggr-config"
+	createPath := client.Endpoint + createHttpUrl
+	createPath = strings.ReplaceAll(createPath, "{project_id}", client.ProjectID)
+	createOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		MoreHeaders:      buildHeaders(d),
+		JSONBody:         utils.RemoveNil(buildMultiAccountAggregationRuleBodyParams(d)),
+	}
+
+	_, err = client.Request("POST", createPath, &createOpt)
+	if err != nil {
+		return diag.Errorf("error creating AOM multi account aggregation rule: %s", err)
+	}
+
+	d.SetId(d.Get("instance_id").(string))
+
+	return resourceMultiAccountAggregationRuleRead(ctx, d, meta)
+}
+
+func buildMultiAccountAggregationRuleBodyParams(d *schema.ResourceData) map[string]interface{} {
+	bodyParams := map[string]interface{}{
+		"dest_prometheus_id": d.Get("instance_id"),
+		"accounts": utils.ValueIgnoreEmpty(buildMultiAccountAggregationRuleAccounts(
+			d.Get("accounts").(*schema.Set).List())),
```

Comment:
buildMultiAccountAggregationRuleAccounts should return nil if no accounts specified
```suggestion
		"accounts": buildMultiAccountAggregationRuleAccounts(
			d.Get("accounts").(*schema.Set).List()),
```